src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs

```diff
@@ -637,7 +637,8 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
                     matchedAccountByLoginHint = wamAccount;
                 }
 
-                if (string.Equals(account.Username, wamAccount.UserName, StringComparison.OrdinalIgnoreCase))
+                if (!string.IsNullOrEmpty(account?.Username) && 
+                    string.Equals(account.Username, wamAccount.UserName, StringComparison.OrdinalIgnoreCase))
                 {
                     matchedAccountByLoginHint = wamAccount;
```

Comment:
nit naming: this is just `matchedAccount` in general, not necessarily by login hint?